### PR TITLE
Case studies carousel

### PR DESCRIPTION
**Pull request by MihneaDinuGL**

### CODE REVIEW

**Review by thomas-ayling — CHANGES_REQUESTED** [review]
- Naming conventions in css leave room for conflicts, instead of things like 'case-study-description' maybe consider adding something like 'case-study-carousel-description' just for complete clarity

- Defining functions in useEffect isn't great, I think you can get away with defining the functions elsewhere and calling them from the useEffect.
- Relating to above, `src/components/case-studies-carousel/BodyCarousel.jsx Lines 10-12` could be removed and you should be able to change lines 14 and 17 out for something like `window.addEventListener('resize', setWindowWidth(window.innerWidth));` Either that or define watchWidth outside of the useEffect

- Try to use arrow functions instead of regular functions just because they are being used in the majority of the code at the moment.

- `src/components/case-studies-carousel/BodyCarousel.jsx Lines 24 - 34, 36-42, 44-50 and 52-62` could be moved into their own components and returned like this https://github.com/thomas-ayling/amcr-front-end/blob/dev/src/components/carousel/shared-components/CarouselCards.jsx
and then implemented like this https://github.com/thomas-ayling/amcr-front-end/blob/dev/src/components/carousel/MainCarousel.jsx (Line 38)

**Review by thomas-ayling — APPROVED** [review]
Changes look

**Review by CSCairney-GL — APPROVED** [review]

File: src/components/case-studies-carousel/BodyCarousel.jsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import { useState, useEffect } from 'react';
+
+import ImageSlider from './sub-components/ImageSlider';
+import TitleSlider from './sub-components/TitleSlider';
+import DescriptionSlider from './sub-components/DescriptionSlider';
+import ButtonSlider from './sub-components/ButtonSlider';
+
+const BodyCarousel = () => {
+  const [current, setCurrent] = useState(0);
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth);
```
Comment (sourcesmith) [inline src/components/case-studies-carousel/BodyCarousel.jsx:12]: Couple of thoughts here:

Resize listeners are a bit dated.  See https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver.  I think there are packages that will provide is a  custom hook or other approaches too.

Why is the carousel not response/styled with breakpoints(media queries)?  Actually I seem some logic around the width to with anchors but I am not quite following it, so is possibly nothing to do with layout.

Comment (MihneaDinuGL) [reply src/components/case-studies-carousel/BodyCarousel.jsx:12]: Hiya, I'll have a look at the resizeobserver and implement something more modern. As for the media queries, I have that at the bottom at the css page, but I'll move it up -- is that what you mean?

Comment (sourcesmith) [reply src/components/case-studies-carousel/BodyCarousel.jsx:12]: Sorry, I am trying to understand why the carousel needs the windowWidth prop.  This suggests something with layout but maybe in this case it is not.

Comment (MihneaDinuGL) [reply src/components/case-studies-carousel/BodyCarousel.jsx:12]: Oh sorry! I just made it so a user can click on the image associated with each case study in the carousel and be redirected to its individual page (same as if using the 'learn more' button). 

However, I wanted to remove this feature for small screens, as users could accidentally click the images and be redirected against their wills, so I just had the windowWidth prop constantly be updated whenever the screen size is changed, and removing that feature whenever the screen gets small enough

Comment (sourcesmith) [reply src/components/case-studies-carousel/BodyCarousel.jsx:12]: Okay, that makes sense.

**Review by sourcesmith — APPROVED** [review]

**Pull request merged**